pytensor/tensor/slinalg.py

```diff
@@ -908,6 +908,101 @@ def solve_discrete_are(A, B, Q, R, enforce_Q_symmetric=False) -> TensorVariable:
     )
 
 
+def largest_common_dtype(tensors: typing.Sequence[TensorVariable]) -> np.dtype:
+    return ft.reduce(lambda l, r: np.promote_types(l, r), [x.dtype for x in tensors])
+
+
+def block_diag_grad(inputs, gout):
```

Comment:
Why did you refactor?

Reply:
This was an intermediate step I didn't delete, i'll take it out.